File: keepr/Models/VaultKeep.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace keepr.Models;

public class VaultKeep : RepoItem<int>
{
  public int VaultId { get; set; }
  public int KeepId { get; set; }
  public string CreatorId { get; set; }
}

public class KeepInVault : Keep
{
  public int VaultId { get; set; }
  public int KeepId { get; set; }
  public int VaultKeepId { get; set; }
}

File: keepr.client/src/models/VaultKeep.js
import { Keep } from "./Keep.js"

export class VaultKeep extends Keep {
  constructor(data) {
    super(data)
    this.vaultId = data.vaultId
    this.keepId = data.keepId
    this.vaultKeepId = data.vaultKeepId
  }
}

// let data = {
//   "vaultId": 78,
//   "keepId": 101,
//   "vaultKeepId": 39,
  
 
//   "id": 101,
//   "createdAt": "2023-08-29T18:31:25",
//   "updatedAt": "2023-08-29T18:31:25"
// }